compiler/parser/parser.peg

```diff
@@ -1118,6 +1118,20 @@ Function
         Loc: loc(c),
       }, nil
     }
+  / !FuncGuard SUBSTRING __ "(" __ e:Expr fromExpr:(_ FROM _ f:Expr { return f, nil })? forExpr:(_ FOR _ f:Expr { return f, nil })? ")" {
```

Comment:
Nit: These things are generally named to match the corresponding fields.
```suggestion
  / !FuncGuard SUBSTRING __ "(" __ expr:Expr from:(_ FROM _ f:Expr { return f, nil })? for_:(_ FOR _ f:Expr { return f, nil })? ")" {
```

Reply:
`for` is a reserved word in go that's I add expr in front of everything for consistency.

Reply:
Then say For or for_?

Reply:
ok